src/extract.jl

```diff
@@ -339,7 +339,7 @@ function read_tarball(
 )
     write_skeleton_header(skeleton, buf=buf)
     # symbols for path types except symlinks store the link
-    paths = Dict{String,Any}()
+    paths = Dict{String,Union{String,Symbol,Int64}}()
```

Comment:
I had Any-typed this to avoid forcing the compiler to over-specialize. Can we just annotate each time we remove something from the dict? Also, this design of using types to differentiate things in this dict is kind of a mess.

Reply:
Yes, we can annotate the retrieved value. Do you want me to do that or are you thinking about other changes instead?